src/tosyntax/FStar.ToSyntax.Interleave.fst

```diff
@@ -231,8 +231,22 @@ let ml_mode_prefix_with_iface_decls
    | Friend _
    | Include _
    | ModuleAbbrev _ ->
-     iface, [impl]
```

Comment:
If we're starting an implementation module, or processing any of its namespace-like directives, take the opens and module decls from the interface as well, and then retain only the vals and tycons from the interface.

This worked better for me, otherwise various compiler modules keep failing in --ide mode.

That said, this whole MLmode interleaving thing is quite a mess. It would be better to just remove interleaving altogether, but that's a much longer discussion